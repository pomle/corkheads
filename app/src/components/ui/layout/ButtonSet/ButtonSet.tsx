import React from "react";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  ButtonSet: {
    display: "grid",
    gridAutoFlow: "column",
    justifyContent: "stretch",
    gridGap: 16,
  },
});

const ButtonSet: React.FC = ({ children }) => {
  const classes = useStyles();

  const nodes = React.Children.toArray(children).filter(
    (node) => node !== null
  );

  if (nodes.length === 0) {
    return null;
  }

  return <div className={classes.ButtonSet}>{nodes}</div>;
};

export default ButtonSet;
