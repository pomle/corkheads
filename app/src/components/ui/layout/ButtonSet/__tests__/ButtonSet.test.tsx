import React from "react";
/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import renderer from "react-test-renderer";
import ButtonSet from "../ButtonSet";

describe("ButtonSet component", () => {
  it("renders nothing if content missing", () => {
    const component = renderer.create(
      <ButtonSet>
        {null}
        {null}
      </ButtonSet>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders content with children when content present", () => {
    const component = renderer.create(
      <ButtonSet>
        <button type="button">Foo</button>
        {null}
        <button type="button">Bar</button>
      </ButtonSet>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
