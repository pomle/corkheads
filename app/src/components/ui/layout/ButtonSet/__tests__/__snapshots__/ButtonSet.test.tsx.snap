// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSet component renders content with children when content present 1`] = `
<div
  className="makeStyles-ButtonSet-1"
>
  <button
    type="button"
  >
    Foo
  </button>
  <button
    type="button"
  >
    Bar
  </button>
</div>
`;

exports[`ButtonSet component renders nothing if content missing 1`] = `null`;
